cunumeric/linalg/linalg.py

```diff
@@ -279,7 +296,13 @@ def _multi_dot_matrix_chain_order(arrays, return_costs=False):
     return (s, m) if return_costs else s
 
 
-def _multi_dot(arrays, order, i, j, out=None):
+def _multi_dot(
+    arrays: Sequence[ndarray],
+    order: Any,
```

Comment:
the type of `order` must be `npt.NDArray[np.int64]`. I think you need an `isinstance(order, npt.NDArray[np.int64])` at the callsite or need to remove `return_costs` from  function `_multi_dot_matrix_chain_order`, as I don't think that function is actually used with `return_costs=True`. (If you decide to do the latter, then the return type of that function will also be simplified.)

Reply:
https://github.com/nv-legate/cunumeric/pull/387/commits/95b3c15f513150053b39a277904816ec562daf2d